girder/cumulus/plugin_tests/volume_test.py

```diff
@@ -692,3 +707,58 @@ def test_get_status(self, get_ec2_client):
             u'status': u'in-use'
         }
         self.assertEqual(r.json, expected, 'Unexpected status')
+
+    @mock.patch('girder.plugins.cumulus.volume.get_ec2_client')
+    @mock.patch('cumulus.ansible.tasks.volume.create_volume.delay')
+    def test_log(self, get_ec2_client, create_volume):
+        volume_id = 'vol-1'
+        ec2_client = get_ec2_client.return_value
+        ec2_client.create_volume.return_value = {
+            'VolumeId': volume_id
+        }
+
+        body = {
+            'name': 'test',
+            'size': 20,
+            'zone': 'us-west-2a',
+            'type': 'ebs',
+            'profileId': self._profile_id
+        }
+
+        r = self.request('/volumes', method='POST',
+                         type='application/json', body=json.dumps(body),
+                         user=self._user)
+        self.assertStatus(r, 201)
+        volume_id = str(r.json['_id'])
+
+        log_entry = {
+            'msg': 'Some message'
+        }
+
+        r = self.request('/volumes/546a1844ff34c70456111185/log', method='GET',
+                         user=self._user)
+        self.assertStatus(r, 404)
+
+        r = self.request('/volumes/%s/log' % str(volume_id), method='POST',
+                         type='application/json', body=json.dumps(log_entry), user=self._user)
+        self.assertStatusOk(r)
+
+        r = self.request('/volumes/%s/log' % str(volume_id), method='GET',
+                         user=self._user)
+        self.assertStatusOk(r)
+        expected_log = {u'log': [{u'msg': u'Some message'}]}
+        self.assertEqual(r.json, expected_log)
+
+        r = self.request('/volumes/%s/log' % str(volume_id), method='POST',
+                         type='application/json', body=json.dumps(log_entry), user=self._user)
+        self.assertStatusOk(r)
+
+        r = self.request('/volumes/%s/log' % str(volume_id), method='GET',
+                         user=self._user)
+        self.assertStatusOk(r)
+        self.assertEqual(len(r.json['log']), 2)
+
+        r = self.request('/volumes/%s/log' % str(volume_id), method='GET',
+                         params={'offset': 1}, user=self._user)
+        self.assertStatusOk(r)
+        self.assertEqual(len(r.json['log']), 1)
```

Comment:
Can we add a test to ensure that the volume log entries appear on the event stream. It should be pretty much like [this test](https://github.com/Kitware/cumulus/blob/master/girder/cumulus/plugin_tests/cluster_test.py#L957-L988)